### PR TITLE
Random Battles updates

### PR DESCRIPTION
**Pull request by ACakeWearingAHat**
-Lucario will now get Inner Focus on the special set.
-Groudon will now always get at least two attacks.
-Palkia and Dialga's Origin formes now hold legal items, and their sets have been changed as such.
-Palkia can now run Choice items.
-Palkia can now run Lustrous Orb.
-Alolan Raichu will now always run a Psychic move.
-Stone Axe counts as having a Stealth Rock user and will prevent future Pokemon from generating Stealth Rock.
-Oinkologne can now run a Liechi Berry sometimes.
-Charizard will no longer get Solar Power on physical sets.
-Spiritomb now always runs Pressure on its Bulky Setup set.
-Sunflora now always runs Chlorophyll because Early Bird is now always rejected.
-Several minor set updates.

### CODE REVIEW

File: data/random-teams.ts [hunk 1]
```diff
@@ -848,8 +848,8 @@ export class RandomTeams {
 		role: string,
 	): boolean {
 		if ([
-			'Armor Tail', 'Battle Bond', 'Flare Boost', 'Gluttony', 'Harvest', 'Hydration', 'Ice Body',
-			'Immunity', 'Own Tempo', 'Quick Feet', 'Rain Dish', 'Snow Cloak', 'Steadfast', 'Steam Engine',
+			'Armor Tail', 'Battle Bond', 'Early Bird', 'Flare Boost', 'Gluttony', 'Harvest', 'Hydration', 'Ice Body',
+			'Immunity', 'Own Tempo', 'Pressure', 'Quick Feet', 'Rain Dish', 'Snow Cloak', 'Steadfast', 'Steam Engine',
```
Comment (MathyFurret) [inline data/random-teams.ts:852]: If the goal was to force Pressure for Bulky Setup Spiritomb, this does quite the opposite

File: data/random-teams.ts [hunk 2]
```diff
@@ -1188,6 +1187,7 @@ export class RandomTeams {
 		if (species.id === 'urshifurapidstrike') return 'Punching Glove';
 		if (species.id === 'lokix' && role === 'Wallbreaker') return 'Life Orb';
 		if (species.id === 'toxtricity' && moves.has('shiftgear')) return 'Throat Spray';
+		if (species.id === 'Palkia') return 'Lustrous Orb';
```
Comment (MathyFurret) [inline data/random-teams.ts:1190]:
```suggestion
		if (species.id === 'palkia') return 'Lustrous Orb';
```

File: data/random-teams.ts (hunk 1 above)
Comment (ACakeWearingAHat) [reply data/random-teams.ts:852]: Forgot to push my branch. This has been fixed.

**Review by MathyFurret — APPROVED** [review]

**Review by livid-washed — APPROVED** [review]

File: data/random-teams.ts [hunk 3]
```diff
@@ -973,6 +971,8 @@ export class RandomTeams {
 		if (abilities.has('Cud Chew') && moves.has('substitute')) return 'Cud Chew';
 		if (abilities.has('Guts') && (moves.has('facade') || moves.has('sleeptalk'))) return 'Guts';
 		if (abilities.has('Harvest') && moves.has('substitute')) return 'Harvest';
+		if (abilities.has('Insomnia') && species.id === 'hypno') return 'Insomnia';
```
Comment (Marty-D) [inline data/random-teams.ts:974]:
```suggestion
		if (species.id === 'hypno') return 'Insomnia';
```

Comment (Marty-D) [reply data/random-teams.ts:974]: Oh it's merged already, never mind then.

**Pull request merged**